helpers/multi.d.ts

```diff
@@ -0,0 +1,9 @@
+declare global {
+    export type MultiReturn<T extends any[]> = T & {
```

Comment:
Looks like intersecting tuple type with anything makes it work incorrectly with destructuring 
```ts
const [_1, _2, _3, _4] = [1, '2', 3];
//                 ^^ Error

declare function $multi<T extends any[]>(...values: T): MultiReturn<T>;

type MultiReturn<T extends any[]> = T;
const [_1, _2, _3, _4] = $multi(1, '2', 3);
//                 ^^ Error

type MultiReturn<T extends any[]> = T & { foo: any };
const [_1, _2, _3, _4] = $multi(1, '2', 3);
//                 ^^ string | number

// Workaround: add an empty tuple
type MultiReturn<T extends any[]> = T & [] & { foo: any };
const [_1, _2, _3, _4] = $multi(1, '2', 3);
//                 ^^ never
```

Reply:
Nice find, thanks for the info!